proposals/0006-Keyboard-Friendly-Virtualization.md

```diff
@@ -0,0 +1,190 @@
+on the rendering of new components. This necessitates a chain of additional
+realized components.
+
+Common keyboarding scenarios should ideally work by default, without having to
+change settings. Some of these scenarios include
+
+1. Navigating from one child component of a rendered item to another
+1. Arrowing between focused items
+1. Shifting focus from currently focused item via page up/down
+1. Shifting focus to first/last item via home/end keys
+
+`VirtualizedList` has enough internal knowledge to transparently support these
+scenarios. This can be accomplished by adding three **"realization windows"**:
+1. **"focused"** Realize cells around a cell with most recent focus. Can be
+tracked by listening to bubbling onFocusCapture in the default
+`CellRendererComponent`.
+1. **"home"** The area at the beginning of the list (for home key).
+    - Note that part of the top may already reamain realized, since the initial
+    render is retained. This may not be the case with non-default
+    `initialItemIndex` however, and the height may not correspond to layout
+    window.
+1. **"end"** The area at the end of the list (for end key).
+
+> Note that Apple devices have "fn + left" or "fn + right" which behave similarly
+> to home/end, but do not move selection in apps like Finder.
+
+The above scenarios are not exhaustive. It may be required to add additional
+APIs allowing external users greater control over realization behavior. These 
+ould potentially be exposed in the future as dynamic/user-controllable
+realization windows, or separate high-level APIs.
+
+### Batch rendering behavior
+
+After rendering a predefined number of initial items, `VirtualizedList` renders
+new items in batches. Batches may not render the full extent of items to
+realize, and the number of items to render per-batch is configurable.
+
+Cells in realization windows should be automatically rendered, even if not yet
+visible. This can be done during batch renders, giving priority to first render
+the window around visible content.
+
+### Endpoint-specific default behaviors
+
+VirtualizedList should be keyboard-friendly by default on desktop platforms as
+they are oriented towards usage of a physical keyboard. This additional
+realization is not free however, increasing memory usage to retain additional
+cells. This tradeoff may not make sense for mobile.
+
+This leads to desired behavior of:
+- **Keep FlatList focus-safe on desktop:** Realization windows needed for
+correct keyboard navigation should be enabled by default.
+- **Keep FlatList memory usage steady on mobile:** New realization windows
+are opt-in where not pay to play.
+
+The full matrix of behaviors is described in the *API Reference* below.
+
+Endpoint detection is generally performed via `Platform` APIs, such as
+`Platform.isTV()` or `Platform.isPad()`. This change would add a
+`Platform.isDesktop()` endpoint query to allow desktop-specific behavior.
```

Comment:
Love this! I [suggest this](https://github.com/facebook/react-native/pull/29937#discussion_r491077211) in facebook/react-native#29937 but that never was completed.

Reply:
I think Android also has this, with`PlatformConstants.constants.uiMode` being 'desk' for desktop.